modules/nf-core/odgi/viz/main.nf

```diff
@@ -0,0 +1,36 @@
+process ODGI_VIZ {
+    tag "$meta.id"
+    label 'process_single'
```

Comment:
Is this the best label to use since you can define the amount of cores used?

Reply:
Ah. If a module can only make use of 1 thread, I enter it directly into the `main.nf`?

Reply:
Some modules are only able to use one core (that's why the `process_single` label exists). I don't know the tool this module is from. Is it a process intensive tool or do you think it will run well using just one module?

The label `process_single` specifies that the module will only get one core so you don't have to define it manually

Reply:
Better now?

Reply:
It can become memory intensive, but it can only make use of a single thread.

Reply:
If the tool can be multithreaded, then allow it to be multithreaded by passing the `$task.cpus`. 
The `process_single` tag has a default setting of 6GB RAM, 1 core. 
The `process_low` tag is 12GB RAM, 2 cores and further scaling occurs with `process_medium` and `process_high`
So if the tool is likely to run out of RAM at 6GB, then default to a larger tag. But definitely using the $task.cpus.

Reply:
So I removed `--threads 1` completely. Should be fine now, right?

Reply:
you should keep `--threads $task.cpus` :) This allows other users to change the amount of cores the program uses

Reply:
But it can only use 1 thread. Dependent on the input it might require several megabytes up to hundreds of gigabytes of RAM.

Reply:
I also implemented parts of `odgi viz` so I know what's going on.

Reply:
The RAM usage can be specified by the user in the configuration file. From what I hear from you, the `process_single` label is the best choice indeed :)

Reply:
> If the tool can be multithreaded, then allow it to be multithreaded by passing the `$task.cpus`. The `process_single` tag has a default setting of 6GB RAM, 1 core. The `process_low` tag is 12GB RAM, 2 cores and further scaling occurs with `process_medium` and `process_high` So if the tool is likely to run out of RAM at 6GB, then default to a larger tag. But definitely using the $task.cpus.

@SPPearce We had the same discussion in https://github.com/nf-core/modules/pull/1149. So we could go down the same route and just keep `process_single`?

Reply:
Yeah, I remember this now. `process_single` but with setting the threads so that the tool will use more if they are available is the way to go.